### DISCUSSION TITLE
styled function issue with Component with as prop

### DISCUSSION THREAD

**Discussion by imshubhamsingh** [chakra-ui/panda#2885] [General]
### Description

styled from jsx ( https://panda-css.com/docs/concepts/style-props#factory-function) function allows you to enable style props to a Component. Issue with this is that if a component also accept `as` prop. The styled function use this `as` prop to render the HTML tag instead of actual Component.

https://github.com/chakra-ui/panda/blob/9da8bbd7814665a4f30ebe7d88813a8af8720f0d/packages/generator/src/artifacts/react-jsx/jsx.ts#L31-L34

For now to get around this problem, I'm managing this using `splitCssProps` directly and not using `styled`
I'm creating a design system using Panda. `as` prop will be a well used prop for primitive components.

We can add few checks to compute what `Element` should be. Below should work, I'm guessing there can be few other cases.
```jsx
     const { children, ...restProps } = props
     const Element = __base__ || props.as
```

I've just started using Panda in production, so far happy with it. If the above solution sounds good, I can also create a fix PR

### Panda CSS Version

0.46.0

**Comment by segunadebayo** [comment]
Do you have a reproduction for the issue? That’ll help me better understand what you mean

**Comment by imshubhamsingh** [comment]
Sure @segunadebayo here is a codesandbox describing the issue https://codesandbox.io/p/devbox/stupefied-mestorf-s5j87f?workspaceId=b4135214-832e-467e-af4e-c0b69cb075ed. In the code, there's a sample component called `Component`, which accepts `text` and `as` as props. Another component, `StyledComponent`, is a styled version of Component, enables html style props. In the App.jsx file, I've used both components in the same way, but the rendered output differs.

I hope this helps.
